support-frontend/assets/components/paymentMethodSelector/paymentMethodSelector.tsx

```diff
@@ -0,0 +1,170 @@
+import { css } from '@emotion/react';
+import type { SerializedStyles } from '@emotion/utils';
+import { from, headline, space } from '@guardian/source-foundations';
+import { Accordion, RadioGroup } from '@guardian/source-react-components';
+import { useEffect } from 'react';
+import GeneralErrorMessage from 'components/generalErrorMessage/generalErrorMessage';
+import { SecureTransactionIndicator } from 'components/secureTransactionIndicator/secureTransactionIndicator';
+import AnimatedDots from 'components/spinners/animatedDots';
+import type {
+	ExistingPaymentMethod,
+	RecentlySignedInExistingPaymentMethod,
+} from 'helpers/forms/existingPaymentMethods/existingPaymentMethods';
+import { mapExistingPaymentMethodToPaymentMethod } from 'helpers/forms/existingPaymentMethods/existingPaymentMethods';
+import type { PaymentMethod } from 'helpers/forms/paymentMethods';
+import { setPaymentMethod } from 'helpers/redux/checkout/payment/paymentMethod/actions';
+import { useContributionsDispatch } from 'helpers/redux/storeHooks';
+import { getReauthenticateUrl } from 'helpers/urls/externalLinks';
+import ContributionChoicesHeader from 'pages/contributions-landing/components/ContributionChoicesHeader';
+import { paymentMethodData } from './paymentMethodData';
+import {
+	AvailablePaymentMethodAccordionRow,
+	ExistingPaymentMethodAccordionRow,
+} from './paymentMethodSelectorAccordionRow';
+
+const container = css`
+	margin-top: ${space[6]}px;
+`;
+
+const header = css`
+	${headline.small({ fontWeight: 'bold' })};
+
+	${from.desktop} {
+		font-size: 28px;
+	}
+`;
+
+function PaymentMethodSelectorLegend() {
+	return (
+		<div
+			css={css`
+				display: flex;
+				justify-content: space-between;
+			`}
+		>
+			<legend id="payment_method">
+				<ContributionChoicesHeader cssOverrides={header}>
+					Payment Method
+				</ContributionChoicesHeader>
+			</legend>
+			<SecureTransactionIndicator position="middle" hideText={true} />
+		</div>
+	);
+}
+
+export interface PaymentMethodSelectorProps {
+	cssOverrides?: SerializedStyles;
+	availablePaymentMethods: PaymentMethod[];
+	paymentMethod: PaymentMethod | null;
+	validationError: string | undefined;
+	fullExistingPaymentMethods?: RecentlySignedInExistingPaymentMethod[];
+	contributionTypeIsRecurring?: boolean;
+	existingPaymentMethod?: RecentlySignedInExistingPaymentMethod;
+	existingPaymentMethods?: ExistingPaymentMethod[];
+}
+
+export function PaymentMethodSelector({
+	availablePaymentMethods,
+	paymentMethod,
+	validationError,
+	fullExistingPaymentMethods,
+	contributionTypeIsRecurring,
+	existingPaymentMethod,
+	existingPaymentMethods,
+}: PaymentMethodSelectorProps): JSX.Element {
+	const dispatch = useContributionsDispatch();
+
+	useEffect(() => {
+		availablePaymentMethods.length === 1 &&
+			dispatch(setPaymentMethod(availablePaymentMethods[0]));
+	}, []);
+
+	const fullExistingRecurringPaymentMethods =
+		contributionTypeIsRecurring && fullExistingPaymentMethods;
```

Comment:
```suggestion
	const fullExistingRecurringPaymentMethods =
		contributionTypeIsRecurring ? fullExistingPaymentMethods : [];
```
Logical AND means this variable could also end up being `false`- usually better to avoid mixing up types too much.